proto/lagrange.proto

```diff
@@ -37,19 +39,34 @@ message WorkerDone {
   }
 }
 
+message UUID {
+  bytes id = 1;
+}
 
 message SubmitTaskRequest {
-  oneof request {
-    string task = 1;
-  }
+  bytes task_bytes = 1;
+  // friendly name of a task for logging and tracing
+  string user_task_id = 2;
+  google.protobuf.Timestamp timeout = 3;
+  bytes price_requested = 4;
+  string class = 5;
 }
 
 message SubmitTaskResponse {
   oneof reply {
     string reply_string = 1;
   }
+  UUID task_uuid = 2;
 }
 
+message ProofChannelRequest {
+  repeated UUID acked_messages = 1;
+  repeated UUID subscribe_to_messages = 2;
```

Comment:
Shouldn't it be an enum ? 
I don't see why we should request a subscription by default ? 
Like if i do `SubmitTask("my_task")` then i expect the GW to automatically return to me the task output corresponding.
If i reconnect and i kept some persistence of the tasks i haven't yet received, then I'll ask to subscribe but why do it by default ?

Reply:
@nikkolasg you can just send an empty vector if you are interested in nothing. the lack of enum here leaves more flexibility: with a single message type you can do two things if you choose so, but with enum you would have to send two messages instead.

Reply:
Okay - on the other hand, it's like sending multiple commands into one now, which is weird imo. We can't extend that struct easily now: one would have to provide a value, even if empty, for all the fields but the one that really matters. 
My main point is we should force a user of this protobuf library to specify stuff he doesn't care. If you're not fan of enum, what do you think then of putting the fields as optionals ?

Reply:
1. it is not possible to put `repeated` fields into `oneof` enum
2. you can not make `repeated` field `optional`, `repeated` field is implicitly optional: you just put nothing into the vector.

Reply:
2. Gotcha.
1. I didn't suggest that - however with enum with you can put other message that themselves contain repeated fields, right ?

Basically there are technical ways to achieve what I'm saying. The question is not technical but from an API perspective.
**Put everything together**: that's the current PR, that forces the user to signal empty fields.
**Separate API** (at the structure / enum level): Where the user only sends what he wants to send.

Sure the separate API requires the user to send a second message if he also wants to re-subscribe to certain tasks IDs but tbh i still don't see the use case for that.
* If client disconnects, then the GW knows that certain set of UUID haven't been acknowledged. 
* When the client reconnects, the GW can directly sent the task outputs for these UUIDs again. No need to re-subscribe.
* Even if he uses this field, it should only be at disconnects no ? 

But fine, let's say we need this call again, and maybe in the future we want more. We don't want to have the user to put values, even if default, on things he doesn't care, that sounds like bad API to me.